signalfx/resource_signalfx_aws_integration_test.go

```diff
@@ -12,148 +13,222 @@ import (
-			namespace = "custom"
-		}
+	custom_namespace_sync_rule {
+	  namespace = "custom"
+	}
 
-		namespace_sync_rule {
-			default_action = "Exclude"
-			filter_action = "Include"
-			filter_source = "filter('code', '200')"
-			namespace = "AWS/EC2"
-		}
-}
+	namespace_sync_rule {
+	  default_action = "Exclude"
+	  filter_action  = "Include"
+	  filter_source  = "filter('code', '200')"
+	  namespace      = "AWS/EC2"
+	}
+  }
 
-resource "signalfx_aws_token_integration" "aws_tok_myteamXX" {
+  resource "signalfx_aws_token_integration" "aws_tok_myteamXX" {
 	name = "AWSFooToken"
-}
+  }
+
+  resource "signalfx_aws_integration" "aws_myteam_tokXX" {
+	enabled = false
+
+	integration_id             = signalfx_aws_token_integration.aws_tok_myteamXX.id
+	token                      = "token123"
+	key                        = "key123"
+	regions                    = ["us-east-1"]
+	poll_rate                  = 300
+	import_cloud_watch         = true
+	enable_aws_usage           = true
+	use_get_metric_data_method = true
+
+	custom_namespace_sync_rule {
+	  default_action = "Exclude"
+	  filter_action  = "Include"
+	  filter_source  = "filter('code', '200')"
+	  namespace      = "fart"
+	}
 
-resource "signalfx_aws_integration" "aws_myteam_tokXX" {
-    enabled = false
-
-		integration_id = "${signalfx_aws_token_integration.aws_tok_myteamXX.id}"
-		token = "token123"
-		key = "key123"
-		regions = ["us-east-1"]
-		poll_rate = 300
-		import_cloud_watch = true
-		enable_aws_usage = true
-		use_get_metric_data_method = true
-
-		custom_namespace_sync_rule {
-			default_action = "Exclude"
-			filter_action = "Include"
-			filter_source = "filter('code', '200')"
-			namespace = "fart"
-		}
+	custom_namespace_sync_rule {
+	  namespace = "custom"
+	}
 
-		custom_namespace_sync_rule {
-			namespace = "custom"
-		}
+	namespace_sync_rule {
+	  default_action = "Exclude"
+	  filter_action  = "Include"
+	  filter_source  = "filter('code', '200')"
+	  namespace      = "AWS/EC2"
+	}
+  }
+`
 
-		namespace_sync_rule {
-			default_action = "Exclude"
-			filter_action = "Include"
-			filter_source = "filter('code', '200')"
-			namespace = "AWS/EC2"
-		}
-}
+const updatedIntegrationAWSConfigMetricStreams = `
+  resource "signalfx_aws_external_integration" "aws_ext_myteamXX" {
+	name = "AWSFoo"
+  }
+
+  resource "signalfx_aws_integration" "aws_myteamXX" {
+	enabled = false
+
+	integration_id     = signalfx_aws_external_integration.aws_ext_myteamXX.id
+	external_id        = signalfx_aws_external_integration.aws_ext_myteamXX.external_id
+	role_arn           = "arn:aws:iam::XXX:role/SignalFx-Read-Role"
+	regions            = ["us-east-1"]
+	poll_rate          = 300
+	import_cloud_watch = true
+	enable_aws_usage   = true
+
+	custom_namespace_sync_rule {
+	  default_action = "Exclude"
+	  filter_action  = "Include"
+	  filter_source  = "filter('code', '200')"
+	  namespace      = "fart"
+	}
+
+	custom_namespace_sync_rule {
+	  namespace = "custom"
+	}
```

Comment:
I suggest to remove both `custom_namespace_sync_rule`s as they are not strictly related to the Metric Streams feature.